packages/ext-taquito/src/tzktReadProvider.ts

```diff
@@ -0,0 +1,169 @@
+import 'cross-fetch/polyfill';
+import {
+  BlockResponse,
+  EntrypointsResponse,
+  MichelsonV1Expression,
+  SaplingDiffResponse, ScriptedContracts,
+} from '@taquito/rpc';
+import {
+  BigMapQuery,
+  BlockIdentifier,
+  SaplingStateQuery,
+  TzReadProvider
+} from '@taquito/taquito';
+import {
+  accountsGetBalance,
+  accountsGetByAddress,
+  accountsGetCounter, bigMapsGetKey, Block, blocksGet,
+  blocksGetByHash,
+  blocksGetByLevel,
+  contractsGetEntrypoints,
+  contractsGetStorage,
+  contractsGetStorageSchema, headGet,
+  protocolsGetCurrent
+} from "@tzkt/sdk-api";
+import {BigNumber} from 'bignumber.js';
+
+
+export class TzktReadProvider implements TzReadProvider {
+  constructor(private readProvider: TzReadProvider) {}
+
+  async getEntrypoints(contract: string): Promise<EntrypointsResponse> {
+    const {data} = await contractsGetEntrypoints(contract);
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    return data;
+  }
+
+  async getScript(address: string): Promise<ScriptedContracts> {
+    const {data} = await contractsGetStorageSchema(address);
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    return data;
+  }
+
+  async getBalance(address: string): Promise<BigNumber> {
+    const {data} = await accountsGetBalance(address);
+    return new BigNumber(data);
+  }
+
+  async getDelegate(address: string): Promise<string | null> {
+    const {data} = await accountsGetByAddress(address);
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    return data?.delegate?.address
+  }
+
+  async getNextProtocol(): Promise<string> {
+    const {data} = await protocolsGetCurrent()
```

Comment:
This will work for all the blocks except for the last block in the protocol (before the upgrade).
Once we have access to tzkt headers we will be able to determine such situations and fallback to the node

Reply:
Need to add block id parameter anyways to implement the interface